packages/serverpod_test/lib/src/serverpod_test.dart

```diff
@@ -0,0 +1,415 @@
+  var rollbackDatabase = maybeRollbackDatabase ?? RollbackDatabase.afterEach;
+  List<InternalTestSession> allTestSessions = [];
+  late _TransactionManager transactionManager;
+
+  return (
+    TestClosure<T> testClosure,
+  ) {
+    const groupDescriptionWithSingleSpaceToEnableIDESupport = ' ';
+    group(groupDescriptionWithSingleSpaceToEnableIDESupport, () {
+      InternalTestSession mainTestSession = InternalTestSession(
+        testServerpod,
+        allTestSessions: allTestSessions,
+        enableLogging: maybeEnableSessionLogging ?? false,
+      );
+
+      setUpAll(() async {
+        await testServerpod.start();
+        var serverpodSession = await testServerpod.createSession();
+        transactionManager = _TransactionManager(serverpodSession);
+
+        if (rollbackDatabase == RollbackDatabase.afterAll ||
+            rollbackDatabase == RollbackDatabase.afterEach) {
+          mainTestSession.transaction =
+              await transactionManager.createTransaction();
+          await transactionManager.pushSavePoint();
+        }
+
+        mainTestSession.setAndConfigureServerpodSession(serverpodSession);
+      });
+
+      tearDown(() async {
+        if (rollbackDatabase == RollbackDatabase.afterEach) {
+          await transactionManager.popSavePoint();
+          await transactionManager.pushSavePoint();
+        }
+
+        if (resetTestSessions == ResetTestSessions.afterEach) {
+          for (var testSession in allTestSessions) {
+            await testSession.resetState();
+          }
+        }
+      });
+
+      tearDownAll(() async {
+        if (rollbackDatabase == RollbackDatabase.afterAll ||
+            rollbackDatabase == RollbackDatabase.afterEach) {
+          await transactionManager.cancelTransaction();
+        }
+
+        for (var testSession in allTestSessions) {
+          await testSession.destroy();
+        }
+        allTestSessions.clear();
+
+        await testServerpod.shutdown();
+      });
+
+      testClosure(testServerpod.testEndpoints, mainTestSession);
+    });
+  };
+}
+
+class _TransactionManager {
+  final List<String> _savePointIds = [];
+
+  Transaction? _transaction;
+
+  late Completer _endTransactionScopeCompleter;
+
+  late Session serverpodSession;
+
+  _TransactionManager(this.serverpodSession);
+
+  Future<Transaction> createTransaction() async {
+    if (_transaction != null) {
+      throw StateError('Transaction already exists.');
+    }
+
+    _endTransactionScopeCompleter = Completer();
+    var transactionStartedCompleter = Completer();
+    late Transaction localTransaction;
+
+    unawaited(
+      serverpodSession.db.transaction((newTransaction) async {
+        localTransaction = newTransaction;
+
+        transactionStartedCompleter.complete();
+
+        await _endTransactionScopeCompleter.future
+            .timeout(const Duration(seconds: 60), onTimeout: () async {
+          await newTransaction.cancel();
+        });
```

Comment:
Question: What is the purpose of placing a timer on the lifetime of a transaction?

The lifetime of a test is configurable. We probably don't want to impose a test time limit with our framework.
Could this be removed? If not, it needs to be be configurable.

Reply:
This can be removed 👍